luarules/gadgets/unit_shipyard_waterline_build.lua

```diff
@@ -0,0 +1,59 @@
+function gadget:GetInfo()
+	return {
+		name = "Shipyards Build at Waterline",
+		desc = "Move the build position depending on unit needs.",
+		author = "robert the pie",
+		layer = 0,
+		enabled = true,
+	}
+end
+
+if not gadgetHandler:IsSyncedCode() then return false end
+
+
+local shipyardsDefIDPads	= {}
+local shipyardsDefIDWaterlines	= {}
+local subDefIDs		= {}
+do
+	local shipyardsNamesToPads = {
+		corsy = 4,
```

Comment:
put piece name as a customparam imo

Reply:
I originally wanted to call the `QueryBuildInfo` but unfortunately: 
1: I couldn't figure out how to get the call to work to get the return param back, if that is even possible.
2: Knowing some of our code malpractices around (me), this could potentially backfire one day.
Will move it to a custom param once i figure out the cases under which the unit goes underground.

Reply:
1) requires LUS afaik, but factories are a great porting target anyway (their script code is simple and they are rare enough for perf not to be a concern)
2) stuff blowing up is good, how else are yall gonna learn?